### ISSUE TITLE
ch[f] in parse_model()

### ISSUE THREAD

**Issue by shihyung** [ultralytics/yolov5#1257]
## ❔Question
I have a question about ch[f] (in parse_model() of yolo.py)
Input parameter "ch" of parse_model()  is a list with value 3 in the begining, which represents the image channel.
So each layer output  channel size c2 appending to "ch" will have an index that is shifted by 1.
For example, ch=[3, 64, 128, 128, 256, 256,... ], and 
index "1" represents output size of [-1, 1, Focus, [64, 3]],       # 0-P1/2, 
index "2" represents output size of [-1, 1, Conv, [128, 3, 2]],  # 1-P2/4, 
... etc.

So the question is:
why c1, c2 = ch[f], args[0] ???
why it is not:    c1 = ch[f if f<0 else f+1]      ??? (especially 'f+1' for index shifted by 1)

it also appears in:
elif m is nn.BatchNorm2d:
            args = [ch[f]]
else:
            c2 = ch[f]

This might cause channel mismatch for self-designed structures with a link from somewhere by ch[f], that it will actually return an output channel size from previous layer.
for example,
[[-1, 1, Focus, [64, 3]],                 # 0-P1/2
   [-1, 1, Conv, [128, 3, 2]],          # 1-P2/4
   [-1, 3, BottleneckCSP, [128]],   # 2
   [-1, 1, Conv, [256, 3, 2]],          # 3-P3/8
   ......
   [3, 1, Conv, [512, 3, 2]],            ---------> what ch[3] actually returns??? 256 or 128 ???

## Additional context
Or maybe I misunderstand this part, and pls forgive my stupid question if so :)
